server/models/ProfileModel.js

```diff
@@ -0,0 +1,59 @@
+const mongoose = require("mongoose");
+Schema = mongoose.Schema;
+const User = require("./User");
+
+const profileSchema = new mongoose.Schema({
+  userId: {
+    type: Schema.Types.ObjectId,
+    ref: "user",
+    required: true,
+  },
+  firstName: {
+    type: String,
+    required: true,
+    minlength: 3,
+    maxlength: 30,
+  },
+  lastName: {
+    type: String,
+    required: true,
+    minlength: 3,
+    maxlength: 30,
+  },
+  address: {
+    type: String,
+    minlength: 3,
+    maxlength: 200,
+  },
+  phone: {
+    type: String,
+    minlength: 5,
+    maxlength: 20,
+  },
+  description: {
+    type: String,
+    minlength: 3,
+    maxlength: 1000,
+  },
+  birth: {
+    type: String,
+    minlength: 3,
+    maxlength: 30,
+  },
+  photos: [String],
+  gender: String,
+  availability: {
+    type: [String],
+    default: [
```

Comment:
I don't think we need a default here as users have the option to manually set their availability instead :)